src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinderProvider.cs

```diff
@@ -43,8 +43,6 @@ public IModelBinder GetBinder(ModelBinderProviderContext context)
             // If the model type is IEnumerable<> then we need to know if we can assign a List<> to it, since
             // that's what we would create. (The cases handled here are IEnumerable<>, IReadOnlyColection<> and
             // IReadOnlyList<>).
-            //
-            // We need to check IsReadOnly because we need to know if we can SET the property.
             var enumerableType = ClosedGenericMatcher.ExtractGenericInterface(modelType, typeof(IEnumerable<>));
             if (enumerableType != null && !context.Metadata.IsReadOnly)
```

Comment:
This change was related to your comment: https://github.com/aspnet/Mvc/pull/5000#discussion_r71980490 ...if i were to remove this check, then both the following would not work...which from my point is totally ok

```
public IEnumerable<> Addresses { get; } = new Address[] { };
public ICollection<> Addresses { get; } = new Address[] { };
```